src/BankID.php

```diff
@@ -75,6 +75,38 @@ public function authenticate($personalNumber, $ip)
         return new BankIDResponse(BankIDResponse::STATUS_PENDING, $httpResponseBody);
     }
 
+    /**
+     * Start a sign-order.
+     *
+     * @param $personalNumber
+     *
+     * @param $ip
+     *
+     * @param $userVisibleData
+     *
+     * @param $userNonVisibleData
+     *
+     * @return BankIDResponse
+     */
+    public function sign($personalNumber, $ip, $userVisibleData = '', $userNonVisibleData = '') {
+        try {
+            $httpResponse = $this->httpClient->post('sign', [
+                RequestOptions::JSON => [
+                    'personalNumber' => $personalNumber,
+                    'endUserIp' => $ip,
+                    'userVisibleData' => $userVisibleData,
+                    'userNonVisibleData' => $userNonVisibleData,
```

Comment:
Since `userNonVisibleData` is optional it should probably only be added to the request if it's not empty (or the default value should be `null`).

Reply:
I agree with you, I changed the default value to `null` and add it to the request if not empty.